src/pages/Login/Login.tsx

```diff
@@ -9,30 +9,56 @@ import {useAppDispatch, useForm} from "../../utils/hooks";
 function Login() {
   const [value, handleChange] = useForm({"E-mail": "", "Password": ""});
   const [type, setType] = React.useState<"password" | "text" | "email">('password');
+  const [error, setError] = React.useState(false);
+  const [isFormValid, setFormValidity] = React.useState<boolean | undefined>(false)
   const dispatch = useAppDispatch();
+  const formRef = useRef<HTMLFormElement | null>(null);
 
+  const keyframes = [
+    {transform: "translateX(0)"},
+    {transform: "translateX(5px)"},
+    {transform: "translateX(-5px)"},
+    {transform: "translateX(5px)"},
+    {transform: "translateX(0)"},
+  ];
 
-  const submitHandler: FormEventHandler = (e) => {
+
+  const submitHandler: FormEventHandler = async (e) => {
     e.preventDefault();
-    dispatch(login(value["E-mail"], value.Password));
+    try {
+      await dispatch(login(value["E-mail"], value.Password))
```

Comment:
NOT implemented: I am using async here to catch the server error and be able to process it accordingly. Admittedly, this may not be the most appropriate approach, but that's the one I came up with

Reply:
ok, we can leave it